Introduction.md

```diff
@@ -123,12 +107,76 @@ This will allow you to run various tools you by name instead of typing the full
 
 ### Updating checkouts
 
-There is a script to update a WebKit checkout: `Tools/Scripts/update-webkit`. This script will automatically merge change logs mentioned below.
+There is a script to update a WebKit checkout: `Tools/Scripts/update-webkit`.
 
 ### Building WebKit
 
 [See Building WebKit](https://github.com/WebKit/webkit/blob/master/ReadMe.md#building-webkit)
 
+### Getting setup to contribute
+
+If you've followed the steps above, get setup to contribute by running:
+
+```Bash
+git webkit setup
+```
+
+The `setup` checks that your environment is optimally configured to contribute, and may prompt you for some additional information.
+
+### Preparing a patch
+
+Firstly, please make sure you [file a bug](https://bugs.webkit.org) for the thing you are adding or fixing! Or, find a bug that you think is relevant to the fix you are making.
+
+Follow the usual conventions for preparing patches in a git workflow:
+
+```Bash
+git checkout -b <branch-name-of-fix-or-feature> main
+```
+
+Once your patch is working, commit them with:
+
+```Bash
+git add <files-to-add>
+git webkit commit -i <bug number from bugs.webkit.org here>
+```
```

Comment:
`git-webkit pr` is able to do all these steps at once and might be more user friendly?

Reply:
Tried this out... but didn't quite work if one is already on a custom feature/bug branch.

For example, doing just `git webkit pr` yields commit message:

```
Need a short description (OOPS!).
Need the bug URL (OOPS!).

Reviewed by NOBODY (OOPS!).

* Introduction.md:
```

And doing, for instance, `git webkit pr --issue 242050` errors with: 

```
Creating a pull-request for '242050' but we're on 'introduction'
```

So, I guess the working assumption is that you start fixing stuff on "main" branch, then do `git webkit pr --issue XXXX` to set up the feature branch and generate the commit log.

Reply:
I also typically work from a local branch and think it would be helpful to include instructions that accommodate this workflow.